### ISSUE TITLE
Access to Interpreter response beyond elements[] upon successful request

### ISSUE THREAD

**Issue by promofu** [plepe/overpass-frontend#2]
I would like to not only gather the feature data (elements[] of response) via callback but also would like to know the osm_base timestamp.

As a temporary solution I added an emit for "response received" right before handlung the results.elements and some req.on(response received) on application side...

If this would be an accepted way, I would be happy to contribute a Pull Request about that.

**Comment by plepe** [comment]
That's an interesting remark. I think it would also be worthwhile to add a value maybe to the `meta` information? Anyway, please submit the pull request, we can discuss matters over there :)

**Comment by promofu** [comment]
ok - I gave it a shot: #3

this one is still lacking proper tests... I couldn't make your tests run because I couldn't easily setup the osm mock backend... maybe some docker magic included in that project would make that easier. I'd volunteer on that if needed ;-)

using the current OSM as a test backend doesn't work out well because of changes that happened after your extract.

**Comment by promofu** [comment]
Using my approach the actual assembly of the "meta" data is left to the consumer like so:

```
req.on('osm3s-results', (result) => {
  osm3sMeta = Object.assign({
    version: result.version,
    copyright: result.copyright,
    generator: result.generator,
  }, result.osm3s)
  console.log(JSON.stringify(osm3sMeta, null, 2))
})
```

we might wrap that into the emit though and make it more standardized ... especially as osm3s seems to add in meta for extensions used in the query (activation of which overpass-frontend controls) -> like AREA

**Comment by plepe** [comment]
To have a docker package with the working OSM test data is a very good idea. I will look into it next week.

I'm not so sure about this event. You are aware, that a request might be splitted into several sub requests? Then the event would be emitted several times. On the other hand, all data might already be cached, then the event would not be emitted at all. Also, simultaneous requests (e.g. one request for shops in the area and a second for restaurants) could be queried at the same time. Then the result contains all elements separated by a "count" element (shops and restaurants) ...

I think it would be better to add this osm3s-meta data as additional property to each feature. Then, even for cached elements, you could see when it was downloaded.

Check out pull request #4, what do you think?

**Comment by promofu** [comment]
I only have a very limited use case where there will be no caching and also I chose a high split number so I wasn't aware of the Subrequest Issue... but sure You are right about it.

**Comment by plepe** [comment]
I was thinking about the issue, and I think it would make sense to use both your code and my code. Because you prefer something on the app level.

As you understood, the 'osm3s-results' event on the request is wrong - but I could move it to the main class. It would make sense there ... So that there's always an event emitted, when some request got data from the server.

Btw, if you set split to 0, the request won't be splitted into sub requests.

**Comment by plepe** [comment]
Merged #4.

**Comment by promofu** [comment]
really sorry for not getting back to you recently on this issue... i will check back as soon as i find some spare time
